src/granite_orm/fields.cr

```diff
@@ -54,6 +54,35 @@ module Granite::ORM::Fields
       return parsed_params
     end
 
+    def to_h
+      fields = {} of String => Bool | Float32 | Float64 | Int32 | Int64 | String | Time | Nil
+
+      {% for name, type in FIELDS %}
+        {% if type.id == Time.id %}
+          fields["{{name}}"] = {{name.id}}.try(&.to_s("%F %X"))
+        {% elsif type.id == Slice.id %}
+          fields["{{name}}"] = {{name.id}}.try(&.to_s(""))
+        {% else %}
+          fields["{{name}}"] = {{name.id}}
+        {% end %}
+      {% end %}
+      {% if SETTINGS[:timestamps] %}
+        if created_at
+          fields["created_at"] = created_at.not_nil!.to_s("%F %X")
+        end
+
+        if updated_at
+          fields["updated_at"] = updated_at.not_nil!.to_s("%F %X")
+        end
```

Comment:
All fields with `nil` values will be present in a hash, but `created_at` and `updated_at` will not. In this intended?

Reply:
@veelenga Good point I'll allow the nil values